app/routes.py

```diff
@@ -1 +1,237 @@
+    goal = validate_model(Goal, goal_id)
+
+    response_body = {
+            "id": goal.id,
+            "title": goal.title,
+            "tasks":[Task.to_dict(task) for task in goal.tasks]
+        }
+    
+    return jsonify(response_body), 200
+
+#************************************************#
+#*************** Helper Functions ***************#
+#************************************************#
+
+def generate_response_body(cls, models):
+    """
+    Return a list of model-detail dictionaries if @param models is a list of Model objects
+    Return a single dictionary {"model": model-detail} if @param models is a Model object
+    """
+    if isinstance(models, list):
+        return [cls.to_dict(model) for model in models]
+    
+    else:
+        return {
+            f"{cls.__name__}".lower(): cls.to_dict(models)
+        }
+
+
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except:
+        response_body = {
+            "message": f"{cls.__name__} id {model_id} is invalid."
+        }
+
+        abort(make_response(jsonify(response_body), 400))
+    
+    model = cls.query.get(model_id)
+
+    if model is None:
+        response_body = {
+            "message": f"{cls.__name__} {model_id} does not exist."
+        }
+
+        abort(make_response(jsonify(response_body), 404))
+
+    return model
+
+
+def get_one_model(cls, model_id):
+    model = validate_model(cls, model_id)
+
+    return jsonify(generate_response_body(cls, model)), 200
+
+
+def delete_one_model(cls, model_id):
+    model = validate_model(cls, model_id)
+
+    db.session.delete(model)
+    db.session.commit()
+
+    response_body = {
+        "details": f"{cls.__name__} {model.id} \"{model.title}\" successfully deleted"
+    }
+
+    return jsonify(response_body), 200
+
+
+def create_one_model(cls, request_body):
+    try:
+        model = cls.create_from_dict(request_body)
+
+        db.session.add(model)
+        db.session.commit()
+
+    except:
+        response_body = {
+            "details": "Invalid data"
+        }
+
+        abort(make_response(jsonify(response_body), 400))
+    
+    return jsonify(generate_response_body(cls, model)), 201
+
+
+def update_one_model(cls, model_id, request_body):
+    model = validate_model(cls, model_id)
+
+    try:
+        cls.update_from_dict(model, request_body)
+
+        db.session.commit()
+    
+    except:
+        response_body = {
+            "details": "Invalid data"
+        }
+
+        abort(make_response(jsonify(response_body), 400))
+
+    return jsonify(generate_response_body(cls, model)), 200
+
+
+def get_all_models(cls, order=None):
+    if order == "asc":
+        models = cls.query.order_by(cls.title).all()
+    elif order == "desc":
+        models = cls.query.order_by(cls.title.desc()).all()
+    else:
+        models = cls.query.all()
+    
+    return jsonify(generate_response_body(cls, models)), 200
+
+
+def post_notification_on_slack(task_title):
```

Comment:
Nice work posting to Slack.  I would suggest maybe making a `post` request instead of a `put` request because you're creating a new message.